lib/WP_Auth0_LoginManager.php

```diff
@@ -587,6 +576,68 @@ public function end_session() {
 		}
 	}
 
+	/**
+	 * Get and filter the scope used for access and ID tokens
+	 *
+	 * @param string $context - how are the scopes being used?
+	 *
+	 * @return string
+	 */
+	public static function get_userinfo_scope( $context = '' ) {
+		return apply_filters( 'auth0_auth_token_scope', 'openid email name nickname picture', $context );
+	}
+
+	/**
+	 * Get a base authorize URL for handling HLP redirects.
+	 *
+	 * @param null|string $connection - specific connection to use, pass null to use all enabled.
+	 * @param null|string $redirect_to - URL to redirect upon successful authentication.
+	 *
+	 * @return array
+	 */
+	public static function get_authorize_params( $connection = null, $redirect_to = null ) {
+		$params = array();
+		$options = WP_Auth0_Options::Instance();
+		$lock_options = new WP_Auth0_Lock10_Options();
+		$is_implicit = (bool) $options->get( 'auth0_implicit_workflow', FALSE );
+		$nonce = WP_Auth0_Nonce_Handler::getInstance()->get();
+
+		$params[ 'client_id' ] = $options->get( 'client_id' );
+		$params[ 'scope' ] = self::get_userinfo_scope( 'authorize_url' );
+		$params[ 'response_type' ] = $is_implicit ? 'id_token': 'code';
+		$params[ 'redirect_uri' ] = $is_implicit
+			? $lock_options->get_implicit_callback_url()
+			: $options->get_wp_auth0_url( null, $is_implicit );
+
+		if ( $is_implicit ) {
+			$params[ 'nonce' ] = $nonce;
+		}
+
+		if ( ! empty( $connection ) ) {
+			$params[ 'connection' ] = $connection;
+		}
+
+		// Get the telemetry header.
+		$telemetry = WP_Auth0_Api_Client::get_info_headers();
+		$params[ 'auth0Client' ] = $telemetry[ 'Auth0-Client' ];
+
+		// Where should the user be redirected after logging in?
+		if ( empty( $redirect_to ) && ! empty( $_GET['redirect_to'] ) ) {
```

Comment:
Line 608 talks about a `redirect_uri`. What would be the difference with this `redirect_to` parameter, and shouldn't they be merged or at least be closer? (lines distance in this file)

Reply:
The `redirect_uri` is the one that /authorize is looking for whereas `redirect_to` is where the plugin sends them after they authenticate (sent with `state` below).